subsys/bluetooth/controller/ll_sw/nordic/lll/lll_df_types.h

```diff
@@ -102,24 +108,33 @@ struct lll_df_sync {
 	struct lll_df_sync_cfg cfg[DOUBLE_BUFFER_SIZE];
 };
 
-/* Names for allowed states for CTE sampling in connected mode */
-enum df_cte_sampling_state {
-	DF_CTE_SAMPLING_UNINITIALIZED,
-	DF_CTE_SAMPLING_ENABLED,
-	DF_CTE_SAMPLING_DISABLED,
-};
-
 /* Names for allowed states for CTE transmit parameters in connected mode */
 enum df_cte_tx_state {
 	DF_CTE_CONN_TX_PARAMS_UNINITIALIZED,
 	DF_CTE_CONN_TX_PARAMS_SET,
 };
 /* Parameters for reception of Constant Tone Extension in connected mode */
 struct lll_df_conn_rx_params {
-	uint8_t state : 2;
-	uint8_t slot_durations : 2; /* One of possible values: 1 us, 2 us. */
-	uint8_t ant_sw_len : 7;
+	uint8_t is_enabled:1;
+	uint8_t ant_sw_len:7;
 	uint8_t ant_ids[BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN];
+	uint8_t slot_durations:2; /* Bit field where: BIT(1) is 1us, BIT(2) is 2us. */
+};
+
+/* Double buffer to store receive and sampling configuration for connected mode */
+struct lll_df_conn_rx_cfg {
+	/* Stores information if the RX configuration was set at least once.
+	 * It is required for handling HCI_LE_Connection_CTE_Request_Enable HCI command.
+	 * See BT 5.3 Core specification Vol 4, Part E, sec. 7.8.85.
+	 */
+	bool is_initialized:1;
```

Comment:
Any reason for `uint8_t is_enabled:1;` but here `bool is_initialized:1;` ? Use `uint8_t is_initialized:1;`

Reply:
If we want to use the same 8-bits for both values, don't we need to do that as well?
i.e.
```
uint8_t is_initialized:1;
uint8_t chan:6;
```

Or can you mix and match `bool` and `uint8_t`?

Reply:
I think that I was asked about such change before. I'm not sure if I were requested to use `bool` in such situation.
Changed to `uint8_t`.

Reply:
I think I requested it to be a bool, but now that `chan` is 6 bits, both of them can fit in a single 8-bit value.

Reply:
> Or can you mix and match bool and uint8_t?

I didn't find clear information to answer the question. I've checked it with gcc 11 and the output was as expected, mix of types didn't increase the size of the structure. 

Ref: https://en.cppreference.com/w/c/language/bit_field